JStanley_Oloughlin_Transmissivity_Calculation

```diff
@@ -0,0 +1,448 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Feb  1 09:42:07 2023
+
+@author: josep
+"""
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import math
+
+def baseflow_filter(y, a, BFImax):
+    """
+    Implementation of the Eckhardt 2005 (HP) recursive digital filter for baseflow separation. Courtesy of Dr. Ciaran Harman. 
+    """
+    N = len(y) # length of the timeseries
+    C = (1 - a) * BFImax / (1 - BFImax)
+    A = a / (1 + C)
+    B = C / (1 + C)
+    b = np.zeros(N)
+    b[0] = BFImax * np.mean(y)
+    for k in range(1,N):
+        if np.isnan(y[k]):
+            b[k] = A * b[k-1]
+        else:
+            b[k] = A * b[k-1] + B * y[k]
+        if b[k]>y[k]:
+            b[k] = y[k]
+    return b
+
+def mouse_event(event):
+    """
+    Parameters
+    ----------
+    event : mouse click
+       This is used so that index points can be found by clicking at any point on the graph.
+
+    Returns
+    -------
+    None.
+
+    """
+    print('x: {} and y: {}'.format(event.xdata, event.ydata))
+        
+def baseflow_calc(Q, P):
+    """
+    Parameters
+    ----------
+    Q : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, discharge (mm/day)
+    P : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, precipitation (mm/day)
+   
+    Returns
+    -------
+     baseflow : Arraay of float64
+        timeseries of baseflow (mm/day) in 15 minute segments in one column 
+
+    """
+    y = Q[6]
+    a = 0.9999
+    BFImax = 0.8
+    baseflow = baseflow_filter(y,a,BFImax)
+    return baseflow
+    
+def hydrograph(Q, P, baseflow):
+    """
+
+    Parameters
+    ----------
+    Q : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, discharge (mm/day)
```

Comment:
why not name the columns their actual names, and then allow the function to look for the actual names rather than the order they are in?